query-object/src/main/java/com/iluwatar/queryobject/App.java

```diff
@@ -0,0 +1,91 @@
+    LOGGER.info("Created dataset with {} customers.", customerDataset.entities.size());
+    final Repository<Customer> customerRepository = new Repository<>(customerDataset.entities);
+    final double purchaseAmount = 200;
+    QueryObject<Customer> queryObject = new CustomerSalesWithPurchaseMoreThan(purchaseAmount);
+    LOGGER.info(
+            "Created query to find all the customers that have spent more than {}.",
+            purchaseAmount);
+
+    Collection<Customer> resultCustomers = customerRepository.query(queryObject);
+    resultCustomers.forEach(
+        res -> LOGGER.info("{} has more than {} purchase amount!", res.name, purchaseAmount)
+    );
+    final var orderNumber = 1;
+    queryObject = new CustomersWithOrdersAmountMoreThan(orderNumber);
+
+    LOGGER.info(
+            "Created query to find all the customers that have made more than {} orders.",
+            orderNumber);
+
+    resultCustomers = customerRepository.query(queryObject);
+    resultCustomers.forEach(
+        res -> LOGGER.info("{} has more than {} orders!", res.name, orderNumber)
+    );
+  }
+
+  private static Dataset<Customer> initializeDataset() {
+    return new Dataset(
+            Arrays.asList(
+                    new Customer(
+                            "Customer 1",
+                            Arrays.asList(
+                                    new CustomerOrder(
+                                            Collections.singletonList(
+                                                    new CustomerOrderDetail(1, 100)
+                                            )
+                                    ),
+                                    new CustomerOrder(
+                                            Arrays.asList(
+                                                    new CustomerOrderDetail(3, 10),
+                                                    new CustomerOrderDetail(4, 20)
+                                            )
+                                    )
+                            )
+                    ),
+                    new Customer(
+                            "Customer 2",
+                            Arrays.asList(
+                                    new CustomerOrder(
+                                            Collections.singletonList(
+                                                    new CustomerOrderDetail(1, 100)
+                                            )
+                                    ),
+                                    new CustomerOrder(
+                                            Arrays.asList(
+                                                    new CustomerOrderDetail(5, 10),
+                                                    new CustomerOrderDetail(2, 20)
+                                            )
+                                    )
+                            )
+                    )
+            )
+    );
+  }

```

Comment:
Consider creating an easier builder-like interface for creating these objects. Instead of customers and orders why not have something fun (dragons, balloons, killer-bees...) to make it more interesting to the readers?